e2e/cypress/integration/multi_team_and_dm/join_open_team_from_dm_spec.js

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// Stage: @prod
+// Group: @multi_team_and_dm
+
+describe('Join an open team from a direct message link', () => {
+    let openTeam;
+    let testUserInOpenTeam;
+
+    let secondTestTeam;
+    let testUserOutsideOpenTeam;
+
+    before(() => {
+        cy.apiCreateTeam('mmt452-second-team', 'mmt452-second-team', 'I', true).then(({team}) => {
+            secondTestTeam = team;
+
+            // # Create pool of users for tests
+            cy.apiCreateUser().then(({user}) => {
+                testUserOutsideOpenTeam = user;
+                cy.apiAddUserToTeam(secondTestTeam.id, testUserOutsideOpenTeam.id);
+            });
+        });
+
+        cy.apiCreateTeam('mmt452-open-team', 'mmt452-open-team', 'O', true).then(({team}) => {
+            openTeam = team;
+
+            // # Allow any user with an account on this server to join this team
+            cy.apiPatchTeam(openTeam.id, {
+                allow_open_invite: true,
+            });
+
+            // # Create pool of users for tests
+            cy.apiCreateUser().then(({user}) => {
+                testUserInOpenTeam = user;
+                cy.apiAddUserToTeam(openTeam.id, testUserInOpenTeam.id);
+
+                // # Login as test user
+                cy.apiLogin(testUserInOpenTeam);
+            });
+        });
+    });
+
+    it('MM-T452 User with no teams should be able to join an open team from a link in direct messages', () => {
+        // # View 'off-topic' channel
+        cy.visit(`/${openTeam.name}/channels/off-topic`);
```

Comment:
Instead of default off-topic channel, please create a new public channel and use it to verify whether the test user also joined the said public channel.